cmd/credentialUtil.go

```diff
@@ -409,7 +409,26 @@ func checkAuthSafeForTarget(ct common.CredentialType, resource, extraSuffixesAAD
 			return fmt.Errorf(
 				"s3 authentication to %s is not currently suported in AzCopy", host)
 		}
+	case common.ECredentialType.GoogleAppCredentials():
+		if resourceType != common.ELocation.GCP() {
+			return errors.New(fmt.Sprintf("Google Application Credentials to %s is not valid", resourceType.String()))
+		}
+
+		ok := false
+		host := "<unparseable url>"
 
+		u, err := url.Parse(resource)
+		if err == nil {
+			host = u.Host
+			parts, err := common.NewGCPURLParts(*u)
+			if err == nil {
+				_, err := url.Parse("https://" + parts.Endpoint)
+				ok = err == nil
+			}
```

Comment:
This is unsafe.

Is there a general format for GCP URIs? Does the GCP SDK have a way to validate a URI is in fact, GCP?

Reply:
Unlike for Azure and AWS where we generate requests depending on the Endpoint provided to us by the user, the EP for GCP is generated by the GCP SDK, based on the bucket Name and objectName (aka object key). Which means that  bucket and object names are only worthwhile info from user input.

In the current code as well, once we've the names, we interact with GCP through the signed URLs generated by SDK. Given that all we need are bucket and object names, appears the SDK does not have a validate an URI.

It is acceptable restriction that we expect the resource to be in format https://storage.cloud.google.com/<bucketName>/<path/to/object>. Taking out EP refs in GCPURLParts, since EP does not make sense here. This restriction means Custom domains and Signed URLs (though used internally) cannot be used. They should be taken at a later stage.